docs/io/misc.rst

```diff
@@ -1,6 +1,6 @@
-*****************************************************
-Miscellaneous: HDF5, YAML, pickle (`astropy.io.misc`)
-*****************************************************
+***********************************************************
+Miscellaneous: HDF5, YAML, ASDF, pickle (`astropy.io.misc`)
```

Comment:
Could you add ``asdf`` to the dependency lists, too? It's scattered all over in pip and rtd requirement files, and narrative docs.

Reply:
It's not a required dependency so I'm not sure where it belongs. Should it just be added to `pip-requirements-dev` and `pip-requirements-doc`?

Reply:
I assume it's the exact same situation as for h5py, the module does some imports of asdf making it an optional dependency overall?

Reply:
Can I do this in a separate PR? Or at least just leave a comment in the requirement file? I'm not sure this makes sense until the corresponding version of `asdf` is released.

Reply:
Sure. I totally missed the conversation above when I left these minor comments, so absolutely address these in a follow-up.